source/compiler/qsc_circuit/src/builder.rs

```diff
@@ -311,6 +367,105 @@ impl CircuitTracer {
         }
         first_user_code_location(&self.user_package_ids, stack)
     }
+
+    fn mark_qubit_in_superposition(&mut self, wire: QubitWire) {
+        self.superposition_qubits.insert(wire.into());
```

Comment:
Perhaps we can have an assert here (and in `flip_classical_qubit` below) that the `prune_classical_qubits` config is set to true. Reading this, I get a bit nervous that we might be doing this work regardless of config flag. I can see you obviously took it into consideration this in this PR, but it feels like it could be accidentally regressed with a future change.